extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/handler/LifxLightHandler.java

```diff
@@ -164,22 +198,23 @@ public void initialize() {
             fadeTime = getFadeTime();
             powerOnBrightness = getPowerOnBrightness();
 
+            channelStates = new HashMap<>();
             currentLightState = new CurrentLightState();
             pendingLightState = new LifxLightState();
 
             communicationHandler = new LifxLightCommunicationHandler(macAddress, currentLightState);
             currentStateUpdater = new LifxLightCurrentStateUpdater(macAddress, currentLightState, communicationHandler,
-                    getThing().getThingTypeUID());
+                    product);
             onlineStateUpdater = new LifxLightOnlineStateUpdater(macAddress, currentLightState, communicationHandler);
-            lightStateChanger = new LifxLightStateChanger(macAddress, pendingLightState, communicationHandler,
+            lightStateChanger = new LifxLightStateChanger(macAddress, pendingLightState, communicationHandler, product,
                     fadeTime);
 
             communicationHandler.start();
             currentStateUpdater.start();
             onlineStateUpdater.start();
             lightStateChanger.start();
         } catch (Exception e) {
-            logger.debug("Error occured while initializing LIFX handler: " + e.getMessage(), e);
+            logger.debug("Error occurred while initializing LIFX handler: " + e.getMessage(), e);
```

Comment:
As you already touched this logging statement, can you fix it and stop using string building (use parameters instead, so the logging framework will build the string only if logging is done).